application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/constrainedtransfer/TransferIndexGenerator.java

```diff
@@ -65,11 +82,13 @@ public ConstrainedTransfersForPatterns generateTransfers() {
       }
 
       try {
-        findTPoints(tx.getFrom(), ALIGHT)
+        findTPoints(tx.getFrom(), ALIGHT, false)
           .stream()
           .filter(TPoint::canAlight)
           .forEachOrdered(fromPoint -> {
-            for (var toPoint : findTPoints(tx.getTo(), BOARD)) {
+            boolean alightFromPreviouslySeenPattern =
+              fromPoint.pattern.patternIndex() <= lastPatternIndex;
```

Comment:
The `pattern.patternIndex() <= lastPatternIndex` is a leaked abstraction, but it is difficult to fix here - I looked into it, but at least you should make sure it is dry - implemented in just one place with a private method:
```
private boolean `isNewPattern(RoutingTripPattern pattern) { return pattern.patternIndex() >= prevProcessedNumberOfPatterns; }`
```
and 
```
boolean alightFromPreviouslySeenPattern = !isNewPattern(fromPoint.pattern);
```